parser.go

```diff
@@ -31,7 +31,7 @@ func tokenEnd(data []byte) int {
 		}
 	}
 
-	return -1
+	return len(data)
```

Comment:
Yes. The "top level" value in a JSON file, as well as being either an `Object` or an `Array` can also be a plain value, like `Number`, `String`, `true`, `false`, or null. I want to use this parser in a project that can parse all JSON files, not only files containing a top level `Object` or `Array`.

Note that all the existing tests pass with this change, and the change (to me) seems logical: the end of a token can be delimited by the end of the string.